src/sentry/static/sentry/app/components/hook.jsx

```diff
@@ -0,0 +1,60 @@
+import PropTypes from 'prop-types';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import HookStore from 'app/stores/hookStore';
+
+/**
+ * Instead of accessing the HookStore directly, use this.
+ *
+ * If the hook slot needs to perform anything w/ the hooks, you can pass a function as a child and you will receive an object with a `hooks` key
+ *
+ * example:
+ *
+ * <Hook name="my-hook">
+ * {({hooks}) => hooks.map(hook => (
+ *  <Wrapper>
+ *    {hook}
+ * </Wrapper>
+ * ))}
+ * </Hook>
+ *
+ */
+const Hook = createReactClass({
+  displayName: 'Hook',
+  propTypes: {
+    name: PropTypes.string.isRequired,
+  },
+  mixins: [Reflux.listenTo(HookStore, 'handleHooks')],
+
+  getInitialState() {
+    let {name, ...props} = this.props;
+
+    return {
+      hooks: HookStore.get(name).map(cb => cb(props)),
```

Comment:
would it be better to apply the props in render?

Reply:
I think it's better to apply them here because you could potentially have hook "sources" that are not components (e.g. returning JSON).